src/Subscriber/CheckoutPageSubscriber.php

```diff
@@ -285,28 +160,51 @@ public function accountPageLoadedEvent( $arg)
     public function paymentMethodPageLoadedEvent($arg)
     {
         $context = $arg->getSalesChannelContext();
-        $isSaveCard = true;
         $isLoggedIn = $context->getCustomer()->getGuest() == true ? false : true;
-        $salesChannelContext = $arg->getSalesChannelContext()->getContext();
         $customField = $context->getCustomer()->getCustomFields();
-        $activeToken = [];
-        
-        // Check if custom field is empty
-        if (!empty( $customField)) {
-            $activeToken = $this->getActiveToken($customField);
-        }
 
         $arg->getPage()->assign(
             [
-                'isSaveCard' => $isSaveCard,
                 'isLoggedIn' => $isLoggedIn,
-                'ckoPaymentMethodId' => $this->getPaymentMethodId($salesChannelContext),
-                'activeToken' => $activeToken,
+                'activeToken' => $this->getActiveToken($customField),
                 'current_page' => 'paymentMethodPageLoadedEvent'
             ]
         );
     }
 
+    /**
+     * validateCutomerInfo
+     *
+     * @param  mixed $customerInfo
+     * @return void
+     */
+    public function validateCutomerInfo($customerInfo)
+    {
+        $info = [];
+
+        if (!empty($customerInfo->getZipCode())) {
```

Comment:
Can we implement `$customerInfo->getAsParam()`?
Which would iterate across internal data an serialize it accordingly.

Reply:
`validateCutomerInfo` to `setCutomerInfo`